app/src/main/java/org/schabi/newpipe/util/StreamItemAdapter.java

```diff
@@ -271,13 +288,145 @@ public static <X extends Stream> Single<Boolean> fetchSizeForWrapper(
+         * @return {@code true} if the {@link MediaFormat} could be retrieved from the response;
+         * otherwise {@code false}
+         * @param <X>
+         */
+        @VisibleForTesting
+        public static <X extends Stream> boolean retrieveMediaFormatFromContentDispositionHeader(
+                @NonNull final X stream,
+                @NonNull final StreamInfoWrapper<X> streamsWrapper,
+                @NonNull final Response response) {
+            // parse the Content-Disposition header,
+            // see
+            // there can be two filename directives
+            String contentDisposition = response.getHeader("Content-Disposition");
+            if (contentDisposition == null) {
+                return false;
+            }
+            try {
+                contentDisposition = Utils.decodeUrlUtf8(contentDisposition);
+                final String[] parts = contentDisposition.split(";");
+                for (String part : parts) {
+                    final String fileName;
+                    part = part.trim();
+
+                    // extract the filename
+                    if (part.startsWith("filename=")) {
+                        // remove directive and decode
+                        fileName = Utils.decodeUrlUtf8(part.substring(9));
+                    } else if (part.startsWith("filename*=")) {
+                        fileName = Utils.decodeUrlUtf8(part.substring(10));
+                    } else {
+                        continue;
+                    }
+
+                    // extract the file extension / suffix
+                    final String[] p = fileName.split("\\.");
+                    String suffix = p[p.length - 1];
+                    if (suffix.endsWith("\"") || suffix.endsWith("'")) {
+                        // remove trailing quotes if present, end index is exclusive
+                        suffix = suffix.substring(0, suffix.length() - 1);
+                    }
+
+                    // get the corresponding media format
+                    final MediaFormat format = MediaFormat.getFromSuffix(suffix);
+                    if (format != null) {
+                        streamsWrapper.setFormat(stream, format);
+                        return true;
+                    }
+                }
+            } catch (final Exception ignored) {
+                // fail silently
+            }
+            return false;
+        }
+
+        @VisibleForTesting
+        public static <X extends Stream> boolean retrieveMediaFormatFromContentTypeHeader(
+                @NonNull final X stream,
+                @NonNull final StreamInfoWrapper<X> streamsWrapper,
+                @NonNull final Response response) {
+            // try to get the format by content type
+            // some mime types are not unique for every format, those are omitted
+            final List<MediaFormat> formats = MediaFormat.getAllFromMimeType(
+                    response.getHeader("Content-Type"));
```

Comment:
`response.getHeader("Content-Type")` will return null if there are no such headers, but fortunately that's not a problem since `getAllFromMimeType` works normally with null strings.